From 072457058d06965a634c35c44b6ecc3c62609e56 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 2 Feb 2025 23:59:57 +0800
Subject: [PATCH 2/2] AOSCOS: bfd: (arm64, powerpc, ppc64, ppc64el) set common
 page size to 64KiB

According to Fedora's binutils spec, "On ppc64 and aarch64 (arm64), we
might use 64KiB pages."

Link: https://src.fedoraproject.org/rpms/binutils/blob/53dfb5f3d99d3ecd5118d5e3957ab48bdc6db74f/f/binutils.spec#_577
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 bfd/elf32-ppc.c     | 2 +-
 bfd/elf64-ppc.c     | 2 +-
 bfd/elfnn-aarch64.c | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/bfd/elf32-ppc.c b/bfd/elf32-ppc.c
index 2ea1574a026..0d74455b066 100644
--- a/bfd/elf32-ppc.c
+++ b/bfd/elf32-ppc.c
@@ -10395,7 +10395,7 @@ ppc_elf_finish_dynamic_sections (bfd *output_bfd,
 #define ELF_TARGET_ID		PPC32_ELF_DATA
 #define ELF_MACHINE_CODE	EM_PPC
 #define ELF_MAXPAGESIZE		0x10000
-#define ELF_COMMONPAGESIZE	0x1000
+#define ELF_COMMONPAGESIZE	0x10000
 #define elf_info_to_howto	ppc_elf_info_to_howto
 
 #ifdef  EM_CYGNUS_POWERPC
diff --git a/bfd/elf64-ppc.c b/bfd/elf64-ppc.c
index 44861a05852..55f37afecce 100644
--- a/bfd/elf64-ppc.c
+++ b/bfd/elf64-ppc.c
@@ -74,7 +74,7 @@ static bfd_vma opd_entry_value
 #define ELF_TARGET_ID		PPC64_ELF_DATA
 #define ELF_MACHINE_CODE	EM_PPC64
 #define ELF_MAXPAGESIZE		0x10000
-#define ELF_COMMONPAGESIZE	0x1000
+#define ELF_COMMONPAGESIZE	0x10000
 #define elf_info_to_howto	ppc64_elf_info_to_howto
 
 #define elf_backend_want_got_sym 0
diff --git a/bfd/elfnn-aarch64.c b/bfd/elfnn-aarch64.c
index 65182f49070..ec33409cd4a 100644
--- a/bfd/elfnn-aarch64.c
+++ b/bfd/elfnn-aarch64.c
@@ -10622,7 +10622,7 @@ const struct elf_size_info elfNN_aarch64_size_info =
 #define ELF_TARGET_ID			AARCH64_ELF_DATA
 #define ELF_MACHINE_CODE		EM_AARCH64
 #define ELF_MAXPAGESIZE			0x10000
-#define ELF_COMMONPAGESIZE		0x1000
+#define ELF_COMMONPAGESIZE		0x10000
 
 #define bfd_elfNN_bfd_is_target_special_symbol	\
   elfNN_aarch64_is_target_special_symbol
-- 
2.48.1

